p2p/peerserver/daghandler.go

```diff
@@ -27,6 +28,11 @@ func (sp *serverPeer) OnSyncResult(p *peer.Peer, msg *message.MsgSyncResult) {
 }
 
 func (sp *serverPeer) OnSyncDAG(p *peer.Peer, msg *message.MsgSyncDAG) {
+	if msg.GS.IsGenesis() && !msg.GS.GetTips().HasOnly(sp.server.chainParams.GenesisHash) {
```

Comment:
Better for a warning msg to show the node maintainer that the genesis hash not match. print out both bad hash & correct hash (`chainParams.GenesisHash`). 
The line without proper logging will result in an obscure warning msg like 
```
Misbehaving peer 127.0.0.1:51969 (inbound): onsyncdag -- ban score increased to 100
Misbehaving peer -- banning and disconnectingpeer="127.0.0.1:51969 (inbound)"
```
This difficult to debug or tracing issue .

Reply:
https://github.com/Qitmeer/qitmeer/pull/255/commits/b8470458f82c4b88dfa42aa2e84fa3b7026b8f20